src/main/java/com/epam/ta/reportportal/core/launch/impl/GetLaunchHandler.java

```diff
@@ -102,116 +103,98 @@ public LaunchResource getLaunch(Long launchId, ReportPortalUser.ProjectDetails p
 		 * condition will be added by server(line 66) Due to limitations of the
 		 * com.mongodb.BasicDBObject, user can't add a second 'mode' criteria
 		 */
-		//		validateModeConditions(filter);
-		//			filter = addLaunchCommonCriteria(DEFAULT, filter, projectName);
+		filter = LaunchFilter.of(filter, LaunchModeEnum.DEFAULT);
 		Page<LaunchFull> launches = launchRepository.findByFilter(ProjectFilter.of(filter, projectName), pageable);
+
+		return PagedResourcesAssembler.pageConverter(LaunchConverter.FULL_TO_RESOURCE).apply(launches);
+	}
+
+	public Iterable<LaunchResource> getDebugLaunches(String projectName, ReportPortalUser user, Filter filter, Pageable pageable) {
```

Comment:
would it make sense to remove this method at all and let user passing debug param they it passes any other filter param? @pbortnik

Reply:
agree. that comment about mode is applied only for mongo db. using postgres we can put this param in one common filter